### PR TITLE
Backend Cooperate Database

### PR DESCRIPTION
**Pull request by fkhan000**
I made a first draft of the cooperate database. I tested it out and I was able to access the database with jdbc. Currently, there are 5 tables: Users, Course, Professor, Reviews, and Likes.

### CODE REVIEW

File: cooperate.sql [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+CREATE SCHEMA public;
+
+DROP DATABASE cooperate;
+CREATE DATABASE cooperate;
+
+\c cooperate;
+--This table contains login info for user as well as their karma
+
+CREATE TABLE Users (
+  id varchar PRIMARY KEY NOT NULL, --username
```
Comment (nicholasingh) [inline cooperate.sql:11]: There doesn't need to be a separate id for each table that's referenced in the review table. We can, for example, make the usernames the primary key for the users and then use that as a foreign key in the review table to save complexity.

File: cooperate.sql [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  PRIMARY KEY(review_id),
+  FOREIGN KEY (course_id) REFERENCES Course,
+  FOREIGN KEY (prof_id) REFERENCES Professor,
+  FOREIGN KEY (user_id) REFERENCES Users
+);
+
+--This table stores the likes and dislikes made by a user for a review
+CREATE TABLE Likes (
+  user_id varchar NOT NULL,
```
Comment (nicholasingh) [inline cooperate.sql:54]: I'm not sure if you need to mention these twice if they're being referenced with the foreign key already. Just saying out of caution

**Review by nicholasingh — CHANGES_REQUESTED** [review]

File: cooperate.sql (hunk 1 above)
Comment (fkhan000) [reply cooperate.sql:11]: I agree, using the foreign keys made the code much more readable and simpler. I added in the changes, please let me know if there are any other issues.

Comment (fkhan000) [reply cooperate.sql:11]: I agree that it isn't really elegant to have to create a column in one line and then declare it as a foreign key in a separate line. Unfortunately, it looks like there isn't any good way to do it in one line in postgres which is pretty strange: https://dba.stackexchange.com/questions/132029/how-to-add-a-column-with-a-foreign-key-constraint-to-a-table-that-already-exists

File: cooperate.sql (hunk 2 above)
Comment (fkhan000) [reply cooperate.sql:54]: I agree that it isn't really elegant to have to create a column in one line and then declare it as a foreign key in a separate line. Unfortunately, it looks like there isn't any good way to do it in one line in postgres which is pretty strange: https://dba.stackexchange.com/questions/132029/how-to-add-a-column-with-a-foreign-key-constraint-to-a-table-that-already-exists

**Pull request merged**